client/txn.go

```diff
@@ -40,34 +40,34 @@ var (
 )
 
 type txnSender struct {
-	*Txn
+	*txn
 }
 
 func (ts *txnSender) Send(ctx context.Context, call Call) {
 	// Send call through wrapped sender.
-	call.Args.Header().Txn = &ts.txn
+	call.Args.Header().Txn = &ts.txn.txn
 	ts.wrapped.Send(ctx, call)
-	ts.txn.Update(call.Reply.Header().Txn)
+	ts.txn.txn.Update(call.Reply.Header().Txn)
 
 	if err, ok := call.Reply.Header().GoError().(*proto.TransactionAbortedError); ok {
 		// On Abort, reset the transaction so we start anew on restart.
-		ts.txn = proto.Transaction{
-			Name:      ts.txn.Name,
-			Isolation: ts.txn.Isolation,
+		ts.txn.txn = proto.Transaction{
+			Name:      ts.txn.txn.Name,
+			Isolation: ts.txn.txn.Isolation,
 			Priority:  err.Txn.Priority, // acts as a minimum priority on restart
 		}
 	}
 }
 
-// Txn provides serial access to a KV store via Run and parallel
+// txn provides serial access to a KV store via Run and parallel
 // access via Prepare and Flush. On receipt of
 // TransactionRestartError, the transaction epoch is incremented and
 // error passed to caller. On receipt of TransactionAbortedError, the
 // transaction is re-created and the error passed to caller.
 //
-// A Txn instance is not thread safe.
-type Txn struct {
-	kv           KV
+// A txn instance is not thread safe.
+type txn struct {
```

Comment:
Yeah, I know this is quite ugly at the moment. I'm planning to merge `txn` into the `Tx` struct and probably rename `Tx` to `Txn`. And as you note, there is lots of other cleanup that can be done here. More PRs are coming.